sql/src/main/java/io/crate/planner/operators/ZeroInputPlan.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.planner.operators;
+
+import io.crate.analyze.relations.AbstractTableRelation;
+import io.crate.analyze.symbol.Symbol;
+
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * A {@link LogicalPlan} with two no other LogicalPlans as input.
+ */
+abstract class ZeroInputPlan extends LogicalPlanBase {
+
+    public ZeroInputPlan(List<Symbol> outputs, List<AbstractTableRelation> baseTables) {
```

Comment:
I would invert the order to match the constructors OneInputPlan and TwoInputPlan

Reply:
Actually, the order is the same as in OneInputPlan and TwoInputPlan. At the moment, it's always 1) source 2) outputs 3) expression mapping 4) base tables 5) dependencies. Here sources are missing, so outputs come first. What we could do is move the outputs always last but it's also kind of nice to have sources and outputs coming first in the constructor.

Reply:
yep it's ok, sorry somehow I mixed up.